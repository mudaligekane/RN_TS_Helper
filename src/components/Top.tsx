import React, {useState} from 'react';

import {Text, ListRenderItemInfo} from 'react-native';
import styled from 'styled-components/native';
import {getUsers} from '../api';
import {Users} from '../types';
import Button from './Button';
import ListItem from './ListItem';

const Container = styled.View`
  flex: 1;
  justify-content: space-between;
  background-color: #f4d8cd;
`;

const T_Input = styled.TextInput.attrs({
  placeholderTextColor: 'yellow',
})`
  height: 50px;
  width: 200px;
  background-color: #1e555c;
  color: white;
`;

const List = styled.FlatList`
background-color:'white
`;

type DeleteReturn = {
  delete:boolean;
}

const Top = () => {
  const [UsersList, SetUsers] = useState<Users[]>([]);

  const GetUsersOnPress = async () => {
    const result = await getUsers();
    if (result.length > 0) {
      SetUsers(result);
    }
  };

  const onPressEdit = (index: string): void => {
    let copyUserList = [...UsersList];
    copyUserList.map((f: Users) => {
      if (f.id === index) {
        f.edit = true;
      }
    });
    SetUsers(copyUserList);
  };



  const onPressDelete = (index: string): DeleteReturn => {
    let copyUserList = [...UsersList];
    let newLi: Users[] = copyUserList.filter((f: Users) => f.id !== index);
    SetUsers(newLi);
    return {delete: true};
  };

  return (
    <Container>
      <Text>Top</Text>

      <Button onPress={GetUsersOnPress} title={'GET Users'} />

      <List
        testID={'List.user'}
        data={UsersList}
        renderItem={({item, index}: ListRenderItemInfo<Users>) => (
          <ListItem
            onPressEdit={onPressEdit}
            onPressDelete={onPressDelete}
            item={item}
            index={index}
          />
        )}
      />
      <T_Input placeholder={'Hello Input'} />
    </Container>
  );
};

export default Top;
