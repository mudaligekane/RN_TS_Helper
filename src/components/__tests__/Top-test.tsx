import React from 'react';
import { render , fireEvent, act} from '@testing-library/react-native';
import axios from 'axios';
import Top from '../Top';

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe('Top Container', () => {
  it('fetch successfully data from API', async () => {
    const mockData = [
      {
        createdAt: '2022-06-17T00:10:47.672Z',
        id: '1',
        name: 'Francis Cole PhD',
      },
      {
        createdAt: '2022-06-17T00:10:47.672Z',
        id: '1',
        name: 'Francis Cole PhD',
      },
    ];

    const { getByTestId , getByText} = render(<Top />);

    mockedAxios.get.mockResolvedValue(mockData);

    const btn = getByText('GET Users');

    fireEvent.press(btn);
  });

  it('to match snapshot', async () => {
    const tree = render(<Top/>);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', async () => {
    const tree = render(<Top/>).toJSON();
    expect(tree.children.length).toBe(4);
  });
});
