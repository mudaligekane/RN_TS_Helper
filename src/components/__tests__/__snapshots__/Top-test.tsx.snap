// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Container to match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f4d8cd",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
      },
    ]
  }
>
  <Text>
    Top
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f15152",
        "borderRadius": 20,
        "height": 40,
        "opacity": 1,
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
        "width": 100,
      }
    }
  >
    <Text>
      GET Users
    </Text>
  </View>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "'white",
        },
      ]
    }
    testID="List.user"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
  <TextInput
    placeholder="Hello Input"
    placeholderTextColor="yellow"
    style={
      Array [
        Object {
          "backgroundColor": "#1e555c",
          "color": "white",
          "height": 50,
          "width": 200,
        },
      ]
    }
  />
</View>
`;
